src/QMCWaveFunctions/tests/test_OptimizableObject.cpp

```diff
@@ -13,16 +13,59 @@
 #include "catch.hpp"
 
 #include "OptimizableObject.h"
+#include "VariableSet.h"
+#include "io/hdf/hdf_archive.h"
 
 namespace qmcplusplus
 {
 
 class FakeOptimizableObject : public OptimizableObject
 {
 public:
-  FakeOptimizableObject(const std::string& my_name) : OptimizableObject(my_name) {}
-  void checkInVariablesExclusive(opt_variables_type& active) override {}
+  FakeOptimizableObject(const std::string& my_name, double var1 = 0.0, double var2 = 0.0) : OptimizableObject(my_name)
+  {
+    myVars.insert("var1", var1);
+    myVars.insert("var2", var2);
+  }
+  void checkInVariablesExclusive(opt_variables_type& active) override { active.insertFrom(myVars); }
+
   void resetParametersExclusive(const opt_variables_type& active) override {}
+
+  void writeVariationalParameters(hdf_archive& hout) override
+  {
+    hout.push("FakeOptimizableObject");
+    // Convert to double before writing to HDF.
+    // If these get written as complex, but read as double, it can cause memory overwrites.
+    // The HDF read call uses the file's type information to determine the amount data read
+    //   (2 doubles for complex), but the variable size is just one double.
+    double var1 = std::real(myVars["var1"]);
+    double var2 = std::real(myVars["var2"]);
+    hout.write(var1, "var1");
+    hout.write(var2, "var2");
+    hout.write(extra_data, "extra_data");
+    hout.pop();
+  }
+
+  void readVariationalParameters(hdf_archive& hin) override
+  {
+    hid_t group = hin.push("FakeOptimizableObject", false);
+    if (group < 0)
+      throw std::runtime_error("Expected group not present in file");
```

Comment:
I feel the error should be handled by `push()`.

Reply:
That might be better, but this PR is not the place to make that change.

Reply:
Could you change 51-53 lines to
```
hin.push("FakeOptimizableObject", false);
```
since we expect to add the guard in `push()` in a later PR.

Reply:
It can be changed later in that PR.  Presumably other places in the code would need a similar change as well.

Reply:
I still feel better to remove the check here and we don't need to revisit this file later. But this PR doesn't need to be blocked.